test/system_test.py

```diff
@@ -74,23 +80,28 @@ def spack_install_and_test(spec: str,
     log_with_spack(f'spack spec {spec}',
                    'system_test',
                    log_filename,
-                   srun=False)
+                   srun=False,
+                   uenv=uenv)
     if split_phases:
         log_with_spack(
             f'spack {command} --until build --test=root -n -v {spec}',
             'system_test',
             log_filename,
-            srun=True)
+            srun=not machine_name() == 'balfrin',
+            uenv=uenv)
         log_with_spack(
             f'spack {command} --dont-restage --test=root -n -v {spec}',
             'system_test',
             log_filename,
-            srun=False)
+            srun=False,
+            uenv=uenv)
     else:
         log_with_spack(f'spack {command} --test=root -n -v {spec}',
                        'system_test',
                        log_filename,
-                       srun=not spec.startswith('icon '))
+                       srun=not spec.startswith('icon')
```

Comment:
"icon<space>" catches "icon" but not "icon-ham".
"icon" catches "icon" and "icon-ham".
Was this change intentional and why?

Reply:
This was by accident, thanks for catching it.